libcalico-go/lib/clientv3/ippool.go

```diff
@@ -296,6 +297,37 @@ func (r ipPools) Watch(ctx context.Context, opts options.ListOptions) (watch.Int
 	return r.client.resources.Watch(ctx, opts, apiv3.KindIPPool, nil)
 }
 
+func equalCIDRStrings(oldCIDR, newCIDR string) bool {
```

Comment:
```suggestion
// cidrChangeOK returns true if the new CIDR is a valid and equivalent version of the old. We allow identical CIDRs. We also allow the new CIDR to be more canonical than the old (but not vice-versa). 
// This is needed because some clients may bypass our validation code that ensures a canonicalized CIDR, and so we must account for unconventional representations being present in our API.
func cidrChangeOK(oldCIDR, newCIDR string) bool {
```

A bit of a nit, but I think the name change is worthwhile since this isn't really an equivalence check - it's checking if thew new CIDR value is allowable based on the current CIDR value.

Also added a comment since this need for this type of logic isn't super intuitive.